samples/apps/copilot-chat-app/webapi/CopilotChat/Models/MemorySource.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Text.Json.Serialization;
+using SemanticKernel.Service.CopilotChat.Storage;
+
+namespace SemanticKernel.Service.CopilotChat.Models;
+
+public enum SourceType
+{
+    File,
+}
+
+/// <summary>
+/// The external memory source.
+/// </summary>
+public class MemorySource : IStorageEntity
+{
+    /// <summary>
+    /// Source ID that is persistent and unique.
+    /// </summary>
+    [JsonPropertyName("id")]
+    public string Id { get; set; } = string.Empty;
+
+    /// <summary>
+    /// The Chat ID.
+    /// </summary>
+    [JsonPropertyName("chatId")]
+    public string ChatId { get; set; } = string.Empty;
+
+    /// <summary>
+    /// The type of the source.
+    /// </summary>
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    [JsonPropertyName("sourceType")]
+    public SourceType SourceType { get; set; } = SourceType.File;
+
+    /// <summary>
+    /// The name of the source.
+    /// </summary>
+    [JsonPropertyName("name")]
+    public string Name { get; set; } = string.Empty;
+
+    /// <summary>
+    /// The external link to the source.
+    /// </summary>
+    [JsonPropertyName("hyperlink")]
+    public Uri? HyperLink { get; set; } = null;
+
+    /// <summary>
+    /// The user name who shared the source.
+    /// </summary>
+    [JsonPropertyName("sharedBy")]
+    public string SharedBy { get; set; } = string.Empty;
+
+    /// <summary>
+    /// When the source is updated in the bot.
+    /// </summary>
+    [JsonPropertyName("updatedOn")]
+    public DateTimeOffset UpdatedOn { get; set; }
```

Comment:
I have a slight preference towards `CreatedOn` because imo it doesn't make sense to have an `UpdateOn` property without a property that tracks the initial creation timestamp and we don't really want / need to track both values. Also, regardless of whether the source already exists in memory, we're creating a new MemorySource object every time a document is imported and overriding the stored value with this new object.

Reply:
However, I don't think the default behavior should be to override an existing memory source if one already exists. I'll talk more to this in a comment on the DocumentImportController file

Reply:
> Member

right this makes sense, regardless of if we override existing ones or not, we're still technically creating a new source every time so `createdOn` does make sense